text/0003-metadata-api.md

```diff
@@ -0,0 +1,62 @@
+- Feature Name: metadata_api
+- Start Date: 2021-07-13
+- RFC PR: (leave this empty)
+- Hathor Issue: (leave this empty)
+- Author: Giovane Costa <gigio.coder@gmail.com>
+
+# Summary
+[summary]: #summary
+
+Metadata API will provide metadata information about Transactions, Tokens, Blocks and other things on Network.
+It will start with Transactions and Tokens
+
+# Motivation
+[motivation]: #motivation
+
+We need to have some data about Transactions, Tokens, Blocks and other things on Network that can't be stored on the
+blockchain. So, we store somewhere else and retrieve them through MetadataAPI
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Information will be stored in S3 in format `(token|transaction)/[id_hash].json` inside the metadata bucket.
+
+A request made to the API passing `id` and `type` will retrieve the information or `404` if not found. 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+**Endpoint:**
+
+GET `/metadata/:type/:id` return `200` | `404`
+
+`metadata-handler` return the stored data
```

Comment:
Maybe we can have a real example with different types of return. One example for a transaction metadata and another one for a token that has NFT data.